iati/validator.py

```diff
@@ -0,0 +1,730 @@
+        err_name = 'err-not-xml-uncategorised-xml-syntax-error'
+
+    error = ValidationError(err_name, locals())
+
+    return error
+
+
+def _create_error_for_rule(rule):
+    """Parse a Rule skip or failure and convert it into an IATI ValidationError.
+
+    Args:
+        rule (iati.rulesets.Rule): The Rule which has either skipped or failed.
+
+    Returns:
+        ValidationError: An IATI ValidationError that contains information about the Rule that has failed.
+
+    """
+    # undertake the mapping between Rule subclass and error name formats
+    rule_to_iati_error_mapping = {
+        'atleast_one': 'err-rule-at-least-one-conformance-fail',
+        'date_order': 'err-rule-date-order-conformance-fail',
+        'dependent': 'err-rule-dependent-conformance-fail',
+        'no_more_than_one': 'err-rule-no-more-than-one-conformance-fail',
+        'regex_matches': 'err-rule-regex-matches-conformance-fail',
+        'regex_no_matches': 'err-rule-regex-no-matches-conformance-fail',
+        'startswith': 'err-rule-starts-with-conformance-fail',
+        'sum': 'err-rule-sum-conformance-fail',
+        'unique': 'err-rule-unique-conformance-fail'
+    }
+
+    try:
+        err_name = rule_to_iati_error_mapping[rule.name]
+    except KeyError:
+        # TODO: it may be desired to make different uncategorised Ruleset errors, depending on findings from usage  # pylint: disable=fixme
+        err_name = 'err-rule-uncategorised-conformance-fail'
+
+    error = ValidationError(err_name, locals())
+
+    return error
+
+
+def full_validation(dataset, schema):
+    """Perform full validation on a Dataset.
+
+    Args:
+        dataset (iati.Dataset): The Dataset to check validity of.
+        schema (iati.Schema): The Schema to validate the Dataset against.
+
+    Warning:
+        Parameters are likely to change in some manner.
+
+    Returns:
+        iati.validator.ValidationErrorLog: A log of the errors that occurred.
+
+    Todo:
+        Create test against a bad Schema.
+
+    """
+    error_log = ValidationErrorLog()
+
+    error_log.extend(_check_is_xml(dataset))
+    error_log.extend(_check_codelist_values(dataset, schema))
+    error_log.extend(_check_ruleset_conformance(dataset, schema))
+
+    return error_log
+
+
+def get_error_codes():
+    """Return a dictionary of the possible error codes and their information.
+
+    Returns:
+        dict: A dictionary of error codes.
+
+    Raises:
+        KeyError: When a specified base_exception is not a valid type of exception.
+
+    Todo:
+        Raise the correct error for incorrect base_exception values.
+        Raise an error when there is a problem with non-base_exception-related errors.
+
+    """
+    err_codes_str = iati.resources.load_as_string(iati.resources.get_lib_data_path('validation_err_codes.yaml'))
+    err_codes_list_of_dict = yaml.safe_load(err_codes_str)
```

Comment:
is it not easier to just to:

`yaml.safe_load(iati.resources.get_lib_data_path('validation_err_codes.yaml'))` ?

Reply:
```
>>> yaml.safe_load(iati.resources.get_lib_data_path('validation_err_codes.yaml'))
'resources/lib_data/validation_err_codes.yaml'
```

Seemingly not